packages/core/src/fetchers/fetchReferencedEntities.ts

```diff
@@ -68,14 +75,68 @@ export const fetchReferencedEntities = async ({
     fetchAllAssets({ client, ids: [...assetIds], locale }),
   ]);
 
+  const usedPatterns = experienceEntry.fields.usedComponents ?? [];
+  const isRenderingExperience = Boolean(!experienceEntry.fields.componentSettings);
+
+  const deepReferences: Array<DeepReference> = gatherDeepReferencesFromExperienceEntry(
+    experienceEntry as ExperienceEntry,
+  );
+
+  // If we are previewing a pattern, we want to include the entry itself as well
+  const fetchedPatterns = (
+    isRenderingExperience ? usedPatterns : [...usedPatterns, experienceEntry]
+  ) as Array<ExperienceEntry>;
+  const allFetchedPatterns = flattenNestedPatterns(fetchedPatterns);
+  const prebindingDataByPatternId = extractPrebindingDataByPatternId(allFetchedPatterns);
+
+  // Patterns do not have dataSource stored in their dataSource field, so head entities won't be there and we need to fetch them
+  if (!isRenderingExperience) {
+    const { dataSource } = generateDefaultDataSourceForPrebindingDefinition(
+      experienceEntry.fields.componentSettings?.prebindingDefinitions,
+    );
+
+    if (Object.keys(dataSource).length) {
+      const prebindingEntriesResponse = await fetchAllEntries({
+        client,
+        ids: Object.values(dataSource).map((link) => link.sys.id),
+        locale,
+      });
+
+      entriesResponse.items.push(...prebindingEntriesResponse.items);
+      entriesResponse.includes.Asset.push(...(prebindingEntriesResponse.includes?.Asset ?? []));
+      entriesResponse.includes.Entry.push(...(prebindingEntriesResponse.includes?.Entry ?? []));
+    }
+  }
+
+  // normally, for experience entry, there should be no need to call this methid, as `includes=2` will have them resolved
+  // because the entires used for pre-binding are stored in both - the layout of the experience, as well as the dataSource field
```

Comment:
```suggestion
  // because the entries used for pre-binding are stored in both - the layout of the experience, as well as the dataSource field
```
nit: noticed in general a couple typos in the comments, maybe worth doing a second pass and fix anything that stands out.